docs/resources/data_source_permission.md

```diff
@@ -42,6 +42,9 @@ resource "grafana_data_source_permission" "fooPermissions" {
     user_id    = 3 // 3 is the admin user in cloud. It can't be queried
     permission = "Query"
   }
+  permissions {
+    permission = "Query"
+  }
```

Comment:
Your assumption is correct, with the Grafana 9.0, when creating a new datasource in Grafana Enterprise (where datasource permissions are available), everyone by default is able to query as the datasource permissions are enabled now by default. 

In addition to that change, there is now Edit permission available next to query (which allows users to edit the datasource). That said, I agree that this datasource needs to be deprecated eventually and we should use the new role-based access control system for managing permissions for datasources. For now, I think this resource is still relevant as you can use it to assign permissions to teams and specific users, in case you take out Query permissions from Viewers, for example. For the way forward I would suggest the following:

- Short term: Add the Edit permission to the allowed list of permissions
- Mid term: Come up with a holistic approach for managing access for DS permissions using RBAC and once there is consensus, mark this resource as a deprecated and start working on the new alternative.

I will create relevant issues to track this.